typing/typecore.ml

```diff
@@ -2988,12 +2988,13 @@ and type_expect_
       assert(is_optional l); (* default allowed only with optional argument *)
       let open Ast_helper in
       let default_loc = default.pexp_loc in
+      let default_ghost = {default.pexp_loc with loc_ghost = true} in
       let scases = [
         Exp.case
-          (Pat.construct ~loc:default_loc
+          (Pat.construct ~loc:default_ghost
              (mknoloc (Longident.(Ldot (Lident "*predef*", "Some"))))
-             (Some ([], Pat.var ~loc:default_loc (mknoloc "*sth*"))))
-          (Exp.ident ~loc:default_loc (mknoloc (Longident.Lident "*sth*")));
+             (Some ([], Pat.var ~loc:default_ghost (mknoloc "*sth*"))))
+          (Exp.ident ~loc:default_ghost (mknoloc (Longident.Lident "*sth*")));
 
         Exp.case
           (Pat.construct ~loc:default_loc
```

Comment:
Is it intentional to keep `default_loc` here? Is it the right choice?

Reply:
Yes, it is intentional: this node corresponds to the default expression in the typedtree (`true` in the example), and I think it is reasonable to leave it to the (non-ghosted) location of the default expression (`true in the example!)

For instance, if we set it a ghost location, I think Merlin would stop helping on e.g. jump to def inside default expressions, as there would be nothing non-ghost at this location:

```ocaml
let f ?(a = let x = 5 in x) = 1
```